[bug] When cross-compiling with android-ndk, CMake cache_variables are not set in the CMakeCache.txt every other run

### Describe the bug

When running with Android NDK the CMake cache_variables aren't set in the CMakeCache.txt every-other time conan is run. This bug appeared in conan 2.7.0, and 2.6.0 does not have the bug.

In this error output, you can see that a version cache_variable was not set, and causes a CMake error. Manual inspection of the CMakeCache.txt shows that the environment variable was never set by conan.
```
[root@caacb8c5e579 sdk]# conan build --build missing --profile:host profiles/android-armv8 --profile:build default --settings build_type=Release .

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.cppstd=gnu17
compiler.libcxx=c++_static
compiler.version=13
os=Android
os.api_level=21
[tool_requires]
*: cmake/3.30.1, android-ndk/r27
[conf]
tools.build:skip_test=True

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=13
os=Linux


======== Computing dependency graph ========
Graph root
    conanfile.py (sdk/1.0.0): /sdk/conanfile.py
Build requirements
    android-ndk/r27#3fd58ec1d8ff51f0aa717227b4404d17 - Cache
    cmake/3.30.1#6d832cf2d46f6ec969ca5ed5b41f91eb - Cache

======== Computing necessary packages ========
Build requirements
    android-ndk/r27#3fd58ec1d8ff51f0aa717227b4404d17:63fead0844576fc02943e16909f08fcdddd6f44b#fbd2d8eaf905d9079b1b6ce862b5cf66 - Cache
    cmake/3.30.1#6d832cf2d46f6ec969ca5ed5b41f91eb:63fead0844576fc02943e16909f08fcdddd6f44b#041a38980949bccd356d9142f4e0a2bf - Cache

======== Installing packages ========

======== Installing packages ========
android-ndk/r27: Already installed! (1 of 2)
cmake/3.30.1: Already installed! (2 of 2)
cmake/3.30.1: Appending PATH environment variable: /sdk/.conan2/p/cmakefaa1321642d5b/p/bin
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'env_info' used in: cmake/3.30.1

======== Finalizing install (deploy, generators) ========
conanfile.py (sdk/1.0.0): Writing generators to /sdk/build/android-armv8/Release/generators
conanfile.py (sdk/1.0.0): Generator 'CMakeDeps' calling 'generate()'
conanfile.py (sdk/1.0.0): Generator 'VirtualBuildEnv' calling 'generate()'
conanfile.py (sdk/1.0.0): Calling generate()
conanfile.py (sdk/1.0.0): Generators folder: /sdk/build/android-armv8/Release/generators
conanfile.py (sdk/1.0.0): CMakeToolchain generated: conan_toolchain.cmake
conanfile.py (sdk/1.0.0): CMakeToolchain: Preset 'conan-android-armv8-release' added to CMakePresets.json.
    (cmake>=3.23) cmake --preset conan-android-armv8-release
    (cmake<3.23) cmake <path> -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE=generators/conan_toolchain.cmake  -DSDK_VERSION=1.0.0 -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DBUILD_TESTING=OFF -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=/sdk/.conan2/p/androaa54feb99bd15/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android21-clang -DCMAKE_CXX_COMPILER=/sdk/.conan2/p/androaa54feb99bd15/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android21-clang++
conanfile.py (sdk/1.0.0): CMakeToolchain generated: /sdk/build/android-armv8/Release/generators/CMakePresets.json
conanfile.py (sdk/1.0.0): CMakeToolchain generated: /sdk/CMakeUserPresets.json
conanfile.py (sdk/1.0.0): Generating aggregated env files
conanfile.py (sdk/1.0.0): Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']

======== Calling build() ========
conanfile.py (sdk/1.0.0): Calling build()
conanfile.py (sdk/1.0.0): Running CMake.configure()
conanfile.py (sdk/1.0.0): RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/sdk" -DSDK_VERSION="1.0.0" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DBUILD_TESTING="OFF" -DCMAKE_BUILD_TYPE="Release" -DCMAKE_C_COMPILER="/sdk/.conan2/p/androaa54feb99bd15/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android21-clang" -DCMAKE_CXX_COMPILER="/sdk/.conan2/p/androaa54feb99bd15/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android21-clang++" "/sdk"
-- Using Conan toolchain: /sdk/build/android-armv8/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting Android platform: android-21
-- Conan toolchain: Setting Android stl: c++_static
-- Conan toolchain: Setting Android abi: arm64-v8a
-- Conan toolchain: C++ Standard 17 with extensions ON
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- Configuring done (0.0s)
You have changed variables that require your cache to be deleted.
Configure will be re-run and you may have to reset some variables.
The following variables have changed:
CMAKE_C_COMPILER= /sdk/.conan2/p/androaa54feb99bd15/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android21-clang
CMAKE_CXX_COMPILER= /sdk/.conan2/p/androaa54feb99bd15/p/bin/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android21-clang++

CMake Error at CMakeLists.txt:9 (project):
  VERSION ".0" format invalid.


-- Configuring incomplete, errors occurred!

ERROR: conanfile.py (sdk/1.0.0): Error in build() method, line 34
	cmake.configure()
	ConanException: Error 1 while executing

```

### How to reproduce it

I have attached a .tar.gz with a basic docker-container reproduction (install docker on x86_64, run a few commands, and you have a repro.)
[repro.tar.gz](https://github.com/user-attachments/files/16931831/repro.tar.gz)

```
# Download repro.tar.gz and extract the reproduction
tar xzvf repro.tar.gz

# On an x86_64 machine, run the following command to create the environment.
docker compose run dev

# Inside the dev machine, run this command to reproduce the issue.
# The issue will reproduce ON THE SECOND RUN. The first command run will work perfectly.
conan build --build missing --profile:host profiles/android-armv8 --profile:build default --settings build_type=Release .
```

This error reproduction is inside a docker container, but it has appeared on bare-metal machines etc. It's not caused by docker.

--- a/conan/tools/cmake/presets.py
+++ b/conan/tools/cmake/presets.py
@@ -5,7 +5,7 @@ import textwrap
 
 from conan.api.output import ConanOutput, Color
 from conan.tools.cmake.layout import get_build_folder_custom_vars
-from conan.tools.cmake.toolchain.blocks import GenericSystemBlock, CompilersBlock
+from conan.tools.cmake.toolchain.blocks import GenericSystemBlock
 from conan.tools.cmake.utils import is_multi_configuration
 from conan.tools.build import build_jobs
 from conan.tools.microsoft import is_msvc
@@ -157,13 +157,6 @@ class _CMakePresets:
                     "value": arch,
                     "strategy": "external"
                 }
-
-        # Set the compiler like in the toolchain. Some IDEs like VS or VSCode require the compiler
-        # being set to cl.exe in order to activate the environment using vcvarsall.bat according to
-        # the toolset and architecture settings.
-        compilers = CompilersBlock.get_compilers(conanfile)
-        for lang, compiler in compilers.items():
-            ret["cacheVariables"][f"CMAKE_{lang}_COMPILER"] = compiler.replace("\\", "/")
 
         ret["toolchainFile"] = toolchain_file
         if conanfile.build_folder:

--- a/conan/tools/cmake/toolchain/blocks.py
+++ b/conan/tools/cmake/toolchain/blocks.py
@@ -865,13 +865,9 @@ class CompilersBlock(Block):
     """)
 
     def context(self):
-        return {"compilers": self.get_compilers(self._conanfile)}
-
-    @staticmethod
-    def get_compilers(conanfile):
         # Reading configuration from "tools.build:compiler_executables" -> {"C": "/usr/bin/gcc"}
-        compilers_by_conf = conanfile.conf.get("tools.build:compiler_executables", default={},
-                                               check_type=dict)
+        compilers_by_conf = self._conanfile.conf.get("tools.build:compiler_executables", default={},
+                                                     check_type=dict)
         # Map the possible languages
         compilers = {}
         # Allowed <LANG> variables (and <LANG>_LAUNCHER)
@@ -882,7 +878,7 @@ class CompilersBlock(Block):
             # To set CMAKE_<LANG>_COMPILER
             if comp in compilers_by_conf:
                 compilers[lang] = compilers_by_conf[comp]
-        return compilers
+        return {"compilers": compilers}
 
 
 class GenericSystemBlock(Block):

--- a/test/integration/toolchains/cmake/test_cmaketoolchain.py
+++ b/test/integration/toolchains/cmake/test_cmaketoolchain.py
@@ -1046,30 +1046,6 @@ def test_set_cmake_lang_compilers_and_launchers():
     assert 'set(CMAKE_RC_COMPILER "C:/local/rc.exe")' in toolchain
 
 
-def test_cmake_presets_compiler():
-    profile = textwrap.dedent(r"""
-    [settings]
-    os=Windows
-    arch=x86_64
-    compiler=msvc
-    compiler.version=193
-    compiler.runtime=dynamic
-    [conf]
-    tools.build:compiler_executables={"c": "cl", "cpp": "cl.exe", "rc": "C:\\local\\rc.exe"}
-    """)
-    client = TestClient()
-    conanfile = GenConanfile().with_settings("os", "arch", "compiler")\
-        .with_generator("CMakeToolchain")
-    client.save({"conanfile.py": conanfile,
-                 "profile": profile})
-    client.run("install . -pr:b profile -pr:h profile")
-    presets = json.loads(client.load("CMakePresets.json"))
-    cache_variables = presets["configurePresets"][0]["cacheVariables"]
-    assert cache_variables["CMAKE_C_COMPILER"] == "cl"
-    assert cache_variables["CMAKE_CXX_COMPILER"] == "cl.exe"
-    assert cache_variables["CMAKE_RC_COMPILER"] == "C:/local/rc.exe"
-
-
 def test_cmake_layout_toolchain_folder():
     """ in single-config generators, the toolchain is a different file per configuration
     https://github.com/conan-io/conan/issues/12827

